cpp/src/arrow/util/value_parsing.h

```diff
@@ -336,6 +439,21 @@ struct StringToSignedIntConverterMixin {
         return false;
       }
     }
+
+    // If its starts with 0x then its hex
+    if (*s == '0' && *(s + 1) == 'x'){
+      length -= 2;
```

Comment:
If there is a negative sign, then it is not a well-formed hex string.

Reply:
Good catch. Fixed this